csrc/ir/utils.cpp

```diff
@@ -191,12 +191,47 @@ struct SubstituteInExpr : public OptOutMutator {
 
 } // namespace ValReplacement
 
-Expr* replaceValInExpr(Expr* expr, Val* reference, Val* substitute) {
+Expr* replaceValInExprInputs(Expr* expr, Val* reference, Val* substitute) {
   FusionGuard fg(expr->fusion());
   return ValReplacement::SubstituteInExpr::subsitute(
       expr, reference, substitute);
 }
 
+Expr* transferDefinitionToNewOutputs(
+    Expr* expr,
+    const std::vector<Val*>& new_outputs) {
+  NVF_ERROR(
+      new_outputs.size() == expr->outputs().size(),
+      "Number of new outputs must match old outputs");
+  OptOutMutator mutator;
+  for (const auto i : c10::irange(new_outputs.size())) {
+    auto old_output = expr->outputs().at(i);
+    auto new_output = new_outputs.at(i);
+    if (new_output == old_output) {
+      continue;
+    }
+    NVF_ERROR(
```

Comment:
Should we also assert that the new outputs are not like `3`?

Reply:
Oh very good idea. This would've helped me find bugs in concretization code sooner.